authors/apps/authentication/serializers.py

```diff
@@ -45,25 +51,26 @@ class RegistrationSerializer(serializers.ModelSerializer):
             )
         ],
         error_messages={
-            'invalid': 'Email must be of the format name@domain.com and should not contain any special characters before @',
+            'invalid': 'Email must be of the format name@domain.com',
             'required': 'Email is a required field'
         }
     )
 
     # Username should be unique
     # Username should not have spaces or special characters
-    # Underscores and hyphens are allowed
+    # Username should be atleast 3 characters and maximum length of 15
+    # Only Underscores and hyphens are allowed
     username = serializers.RegexField(
-        regex='^(?!.*\ )[A-Za-z\d\-\_]+$',
+        regex='^[a-z0-9._-]{3,15}$',
         required=True,
         validators=[
             UniqueValidator(
                 queryset=User.objects.all(),
-                message='username already taken',
+                message='user with this username already exists',
             )
         ],
         error_messages={
-            'invalid': 'Username should have no spaces or special characters only',
+            'invalid': 'Username should be at least 3 characters, have no spaces or special characters except ._-',
```

Comment:
Also make the accepted special characters here clearer, right now one can't tell which are the accepted ones. Try naming the accepted ones and enclosing the character in a bracket as such ``hyphen (-), fullstop (.), e.t.c``

Reply:
Hello @IssaIan ,
I think we could let our users choose any special character. That would help make their passwords unique. [special characters](https://www.owasp.org/index.php/Password_special_characters)

Reply:
@Ogutu-Brian That's the username

Reply:
I thought it was the password